File: src/main/java/sprykerFileCreator/model/definition/spryker/ZedQueryContainerDefinition.java
package pav.sprykerFileCreator.model.definition.spryker;

import pav.sprykerFileCreator.model.definition.ClassDefinitionInterface;
import pav.sprykerFileCreator.model.generator.SprykerConstants;

public class ZedQueryContainerDefinition implements ClassDefinitionInterface {

    @Override
    public String getClassType() {
        return SprykerConstants.ZED_QUERY_CONTAINER;
    }

    @Override
    public String getNamePattern() {
        return "{{ bundleName }}QueryContainer";
    }

    @Override
    public String getNamespacePattern() {
        return "{{ projectName }}\\Zed\\{{ bundleName }}\\Persistence";
    }

    @Override
    public String getMethodForReturnType() {
        return "getQueryContainer()";
    }

    @Override
    public String[] getDocBlockClasses() {
        String [] docBlockClasses = {
                SprykerConstants.ZED_PERSISTENCE_FACTORY
        };

        return docBlockClasses;
    }

    @Override
    public String getDefaultParentClass() {
        return "Spryker\\Zed\\Kernel\\Persistence\\AbstractQueryContainer";
    }
}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/definitions/spryker/ZedQueryContainerDefinition.kt
package com.github.tobi812.sprykerplugin.models.definitions.spryker

import com.github.tobi812.sprykerplugin.constants.SprykerConstants
import com.github.tobi812.sprykerplugin.models.definitions.ClassDefinitionInterface

class ZedQueryContainerDefinition : ClassDefinitionInterface {
    override val classType: String
        get() = SprykerConstants.ZED_QUERY_CONTAINER
    override val namePattern: String
        get() = "{{ bundleName }}QueryContainer"
    override val namespacePattern: String
        get() = "{{ projectName }}\\Zed\\{{ bundleName }}\\Persistence"
    override val methodForReturnType: String
        get() = "getQueryContainer()"
    override val docBlockClasses: Array<String>
        get() = arrayOf()
    override val defaultParentClass: String
        get() = "Spryker\\Zed\\Kernel\\Persistence\\AbstractQueryContainer"
}
